C/kmsg_log.c

```diff
@@ -51,32 +51,59 @@ int klog(char *msg)
 	return ret;
 }
 
-	int logwarn(char *warning) 
-	{ 
-		char *warn ="Warning! : "  ;
-		strcat(*warn,*warning);
-		int ret = klog(*warn);
-		return(ret);
-	}	
-
-	int logerror(char *error) 
-	{ 
-		char *error_tag ="Error! : "  ;
-		strcat(*error_tag,error);
-		int ret = klog(*error_tag);
-		return(ret);
-	}	
-	int logdebug (char *debug_msg)
-	{
-		char *debug_tag = "Debug : ";
-		strcat(*debug_tag,debug_msg);
-		int ret = klog(*debug_tag);
-		return(ret);
-	}
-	int loginfo (char *info)
-	{
-		char *info_tag = "Info : ";
-		strcat(*info_tag,info);
-		int ret = klog(*info_tag);
-		return(ret);
-	}
+/*
+  This functions prefixes the Warning label
+  to the real message and sends it to klog
+  for real "printing" to kmsg (IF LOG_LOC not defined or used)
+*/
+int logwarn(char *warning)
+{
+ char *warn = malloc(256);
+  strcpy(warn ,"Warning!: ");
+  strcat(warn ,warning);
+ int ret = klog(warn) ;
+  return(ret);
+  free(warn);
+ }
+  /*
+    This functions prefixes the Error label
+    to the real message and sends it to klog
+    for real "printing" to kmsg (IF LOG_LOC not defined or used)
+  */
+int logerror(char *error_msg)
+{
+ char *error = malloc(256);
+  strcpy(error,"Error!: ");
+  strcat(error,error_msg);
+ int ret = klog(error);
+  return(ret);
+  free(error);
+ }
+  /*
+    This functions prefixes the debug label
+    to the real message and sends it to klog
+    for real "printing" to kmsg (IF LOG_LOC not defined or used)
+  */
+int logdebug (char *debug_msg)
+{
+ char *debug = malloc(256);
```

Comment:
Your are using malloc everywhere, good catch. 
But you set the same value on all entries. Can you use a preprocessor variable instead please?